src/wasm.rs

```diff
@@ -610,6 +580,75 @@ where
         }
     }
 
+    /// Processes WasmMsg::Instantiate and WasmMsg::Instantiate2 messages.
+    fn process_wasm_msg_instantiate(
+        &self,
+        api: &dyn Api,
+        storage: &mut dyn Storage,
+        router: &dyn CosmosRouter<ExecC = ExecC, QueryC = QueryC>,
+        block: &BlockInfo,
+        sender: Addr,
+        admin: Option<String>,
+        code_id: u64,
+        msg: Binary,
+        funds: Vec<Coin>,
+        label: String,
+        _salt: Option<Binary>,
```

Comment:
I suppose this will be used when the [new canonicalize and humanize](https://github.com/CosmWasm/cosmwasm/issues/1873) are implemented

Reply:
Yes, definetely. When the `MockApi` has new versions of `addr_canonicalize` and `addr_humanize` that are compatible with `instantiate2_addr` function, than the salt, contract's checksum and the creator address will be used to generate the predictable address, exactly like it is done in real-life blockchain.